oauth-common/src/main/java/io/strimzi/kafka/oauth/common/IOTask.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright 2017-2023, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.common;
+
+import java.io.IOException;
+
+public interface IOTask<T> {
```

Comment:
Can we have a sentence of Javadoc to explain what this interface abstracts, and exactly what the contract is. For example, I assume there's an assumption somewhere that `run()` should be idempotent. Otherwise if we ended up re-attempting a request which had actually already succeeded in a prior request we'd incur the same non-idempotent side-effects.

Also it seems a bit odd to have `IOTask` but the retry method be in `HttpUtil`. The retry is actually independent of HTTP AFAICS, so you could make it a static method of this class. Or perhaps just rename it `HttpTask`.

Reply:
You are right, this interface should be better named as `HttpTask` as it exists to perform HTTP requests to the authorization server. It primarily exist because the `run()` method on it can throw an `IOException` which makes the code using it more elegant, not adding another needless layer of additional exception handling, unwrapping from a `RuntimeException`, difficulty to differentiate between wrapping and non-wrapped `RuntimeException`, introducing suttle differences of behaviour between invoking `HttpUtil` methods directly or via `HttpUtil.doWithRetries()`.

Reply:
I renamed IOTask to HttpTasks.